src/serde.rs

```diff
@@ -0,0 +1,119 @@
+use core::str::FromStr;
+use serde::{
+    de::{self, Deserializer, Visitor},
+    // ser::Serializer,
+    Deserialize, //Serialize,
+};
+use std::fmt;
+
+impl<'de> Deserialize<'de> for crate::Date {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(DateVisitor)
+    }
+}
+
+struct DateVisitor;
+impl<'de> Visitor<'de> for DateVisitor {
+    type Value = crate::Date;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a string that follows iso8691 Date format")
```

Comment:
```suggestion
        write!(formatter, "a string that follows ISO8601 date format")
```

8691? what are you trying to set on fire?

Reply:
OMG, LOL, that's why I don't merge my late night commits myself 🤣